cmd/kubeadm/app/phases/addons/manifests.go

```diff
@@ -68,7 +68,7 @@ spec:
     spec:
       containers:
       - name: kube-proxy
-        image: {{ .Image }}
+        image: {{ if .ImageOverride }}{{ .ImageOverride }}{{ else }}{{ .ImageRepository }}/kube-proxy-{{ .Arch }}:{{ .Version }}{{ end }}
```

Comment:
for now, I think it's better to be consistent with current behavior, same as with GetCoreImage()
Altering that functionality better to be in own PR with separate changelog.